tests/policy_as_code_test.go

```diff
@@ -0,0 +1,239 @@
+	ch := make(chan *v1alpha1.SecurityPolicy)
+	reg, err := pc.informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			p := pc.toPolicy(newObj)
+			ch <- p
+		},
+	})
+	pc.Require().NoError(err)
+
+	deferFunc := func() {
+		pc.Require().NoError(pc.informer.Informer().RemoveEventHandler(reg))
+		close(ch)
+	}
+	return ch, deferFunc
+}
+
+func (pc *PolicyAsCodeSuite) toPolicy(i interface{}) *v1alpha1.SecurityPolicy {
+	policyCR := v1alpha1.SecurityPolicy{}
+	obj := i.(*unstructured.Unstructured)
+	pc.Require().NoError(runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &policyCR))
+	return &policyCR
+}
+
+func (pc *PolicyAsCodeSuite) toUnstructured(i interface{}) *unstructured.Unstructured {
+	m, err := runtime.DefaultUnstructuredConverter.ToUnstructured(i)
+	pc.Require().NoError(err)
+	return &unstructured.Unstructured{Object: m}
+}
+
+func (pc *PolicyAsCodeSuite) checkPolicyIsDeclarative(id string) {
+	pc.Require().EventuallyWithT(func(collect *assert.CollectT) {
+		policy, err := pc.centralClient.GetPolicy(pc.ctx, &v1.ResourceByID{Id: id})
+		if err != nil {
+			collect.Errorf("Failed to get policy: %s", err.Error())
+		}
+		if policy.Source != storage.PolicySource_DECLARATIVE {
+			collect.Errorf("Policy %s was not marked as declarative in Central", id)
+		}
+	}, time.Second*5, time.Millisecond*30)
+}
+
+func (pc *PolicyAsCodeSuite) updateCRandObserveInCentral(k8sPolicy *v1alpha1.SecurityPolicy, id string) {
+	k8sPolicy.Spec.PolicySections[0].PolicyGroups[0].Values[0].Value = "3"
+	_, err := pc.k8sClient.Update(pc.ctx, pc.toUnstructured(k8sPolicy), metav1.UpdateOptions{})
+	pc.Require().NoError(err)
+
+	pc.Require().EventuallyWithT(func(collect *assert.CollectT) {
+		policy, err := pc.centralClient.GetPolicy(pc.ctx, &v1.ResourceByID{Id: id})
+		if err != nil {
+			collect.Errorf("Failed to get policy: %s", err.Error())
+		}
+		criteriaValue := policy.PolicySections[0].PolicyGroups[0].Values[0].Value
+		if criteriaValue != "3" {
+			collect.Errorf("Policy criteria not updated in Central. Expected 3 but got %s", criteriaValue)
+		}
+	}, time.Second*5, time.Millisecond*30)
+}
+
+func (pc *PolicyAsCodeSuite) deleteCRandObserveInCentral(k8sPolicy *v1alpha1.SecurityPolicy, id string) {
+	pc.Require().NoError(pc.k8sClient.Delete(pc.ctx, k8sPolicy.GetName(), metav1.DeleteOptions{}))
+
+	pc.Require().EventuallyWithT(func(collect *assert.CollectT) {
+		_, err := pc.centralClient.GetPolicy(pc.ctx, &v1.ResourceByID{Id: id})
+		if err != nil {
+			statusErr, _ := status.FromError(err)
+			if statusErr.Code() != codes.NotFound {
+				collect.Errorf("Unexpected error when geting policy: %s", err.Error())
+			}
+		} else {
+			collect.Errorf("Successfully fetched policy %s when it should be deleted", id)
+		}
+	}, time.Second*5, time.Millisecond*30, "Policy CR deletion not propogated to Central")
+}
+
+func (pc *PolicyAsCodeSuite) TearDownSuite() {
+	// TODO: Don't double delete
```

Comment:
nit:  Still valid?

Reply:
Yes, I think so.  This issues a delete to Central for all policies created, but it should have already been deleted by the config-controller when the corresponding CR was deleted (we in fact validate that the policy gets deleted from Central).  So this is a "double delete" when the test succeeds, but it still needs to be here for when the test fails halfway through.